src/hocon_schema.erl

```diff
@@ -194,13 +201,23 @@ find_structs(Schema) ->
 do_roots(Mod) when is_atom(Mod) -> Mod:roots();
 do_roots(#{roots := Names}) -> Names.
 
+%% @doc Get fields of the struct for the given struct name.
 -spec fields(schema(), name()) -> [field()].
-fields(Mod, Name) when is_atom(Mod) ->
-    Mod:fields(Name);
-fields(#{fields := Fields}, Name) when is_function(Fields) ->
-    Fields(Name);
-fields(#{fields := Fields}, Name) when is_map(Fields) ->
-    maps:get(Name, Fields).
+fields(Sc, Name) ->
```

Comment:
if we only return fields meta by `fields/1`, then the `find_struct/1` can't be private. 
I need the get struct from emqx_dashboard apps to generate API.

so `find_struct/1` should be public or make fields/1 return all meta(bad idea...)

Reply:
There is fields_and_meta/1

Reply:
Exported